helper/schema/grpc_provider.go

```diff
@@ -970,6 +971,24 @@ func (s *GRPCProviderServer) ReadResource(ctx context.Context, req *tfprotov5.Re
 			return resp, nil
 		}
 
+		if !res.ResourceBehavior.AllowNullIdentity {
```

Comment:
nit: In read this check happens after the mutable identity check, in apply it happens before. Should they be consistent (either way)